attributes/default.rb

```diff
@@ -20,7 +20,16 @@
                                            else
                                              '/etc/consul-template.d'
                                            end
-default['consul_template']['init_style'] = platform?("ubuntu") ? 'upstart' : 'init' # 'init', 'runit', 'systemd', 'upstart'
+# 'init', 'runit', 'systemd', 'upstart'
+default['consul_template']['init_style'] = if platform?("ubuntu")
+  if Chef::VersionConstraint.new('>= 15.04').include?(node['platform_version'])
+    'systemd'
+  else
+    'upstart'
+  end
+else
+  'init'
+end
```

Comment:
Not sure which version this was introduced in but it might be easier/simpler to use `node['init_package']`.